FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -29,14 +29,49 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		let context = self.context
+		context.perform {
```

Comment:
Suggestion: another way of capturing the `context` inside the closure without capturing `self` is to use a capture list as shown below:

```suggestion
		context.perform { [context] in
```

It achieves the same result with one less line of code. Feel free to use it in all methods that call `context.perform`.